crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -577,15 +610,29 @@ impl WritebackCache {
         epoch: EpochId,
         digest: TransactionDigest,
     ) -> SuiResult {
-        let Some((_, outputs)) = self.dirty.pending_transaction_writes.remove(&digest) else {
+        let DashMapEntry::Occupied(occupied) = self.dirty.pending_transaction_writes.entry(digest)
+        else {
             panic!("Attempt to commit unknown transaction {:?}", digest);
         };
 
-        // Flush writes to disk
+        let outputs = occupied.get().clone();
+
+        // Flush writes to disk before removing anything from dirty set. otherwise,
+        // a cache eviction could cause a value to disappear briefly, even if we insert to the
+        // cache before removing from the dirty set.
         self.store
             .write_transaction_outputs(epoch, outputs.clone())
```

Comment:
Do we want to make DB operations used by the cache async, so that we can inject faults before/after and catch bugs in simtests?